src/dealDashboard/dealInfo/dealInfo.ts

```diff
@@ -2,12 +2,47 @@ import { autoinject, bindable } from "aurelia-framework";
 import { DealTokenSwap } from "entities/DealTokenSwap";
 import { availableSocialMedias } from "wizards/tokenSwapDealWizard/dealWizardTypes";
 import "./dealInfo.scss";
+import { DisposableCollection } from "../../services/DisposableCollection";
+import { EventAggregator } from "aurelia-event-aggregator";
+import { AureliaHelperService } from "../../services/AureliaHelperService";
 
 @autoinject
 export class DealInfo {
   @bindable deal: DealTokenSwap;
 
+  private subscriptions = new DisposableCollection();
+  private settingPrivacy = false;
+
+  constructor(
+    private eventAggregator: EventAggregator,
+    private aureliaHelperService: AureliaHelperService,
+  ) {
+  }
+
   getSocialMediaDetails(socialMediaName: string) {
     return availableSocialMedias.find(socialMedia => socialMedia.name === socialMediaName);
   }
+
+  public async attached() {
+    this.subscriptions.push(
+      this.aureliaHelperService.createPropertyWatch(this.deal, "isPrivate", async (value: boolean) => {
```

Comment:
Can we remove this subscription and either:

1) have a method in this viewmodel that is executed on click of the toggle and does therein whatever is needed, or better:
2) have the toggle support a bindable click method that is invoked on click and does therein whatever is needed.

I vote #2.